### ISSUE TITLE
[DX] Check in the App/ folder for DataFixtures/ORM (or MongoDB)

### ISSUE THREAD

**Issue by parhamdoustdar** [doctrine/DoctrineFixturesBundle#137]
Since Symfony 2.6 has moved to application-wide code (such as migrations, configuration, etc) living in the `App/` folder, it will improve the developer experience if this bundle also checked the `App/` folder prior to looking inside different bundles, since from now on, applications will mostly have only one bundle.

**Comment by lavoiesl** [comment]
AFAIK, _migrations_ are stored in `app/`, but Fixtures are still stored in bundles. The best practice suggests to create a single bundle, `AppBundle`, which means Fixtures would still be in `src/AppBundle/DataFixtures`. http://symfony.com/doc/current/best_practices/creating-the-project.html#application-bundles

**Comment by parhamdoustdar** [comment]
True, @lavoiesl. I wasn't saying that there will be no bundles from here on out. What I was saying is that the `app/` directory is getting more attention due to this change. For example, the templates are now recommended to go to `App/Resources/Views`, and migrations are in there by default, too, so it only makes sense for the `app/` directory to at least be checked in the process of finding migrations, too.

The benefits of having fixtures inside bundles really shines when you have multiple bundles, each with their very own fixture. However, for those who have only one bundle, and are not planning to reuse this bundle anywhere, and are only using it to write a custom application of their own, the `DataFixturesBundle` is seemingly breaking the convention.

At least, for a newcomer to the framework, having bundles that each put their folders in a different location is very confusing. It gives the sense that the differences between the `app/` folder and the `src/` folder are not very well defined.

I hope I have been able to explain my case. :)

**Comment by lavoiesl** [comment]
I would agree with you as soon as entities are moved to `app/`. In the mean time, it seems weird to have the entities at one place and the fixtures at another. I know @weaverryan once did a quick hack that puts almost all the `AppBundle` inside `app/`, but it was simply a proof of concept. 

For this to have more traction, I would consider creating a more general issue in Symfony and include the rest of AppBundle like controllers, entities, services, etc.

**Comment by lavoiesl** [comment]
By the way, migrations are not connected to ORM/Entities, they act on the whole database, it would make no sense to include them in a bundle, it is necessarily applying to the whole project. Therefore, they are at the proper place, I can’t say the same for the fixtures.

**Comment by parhamdoustdar** [comment]
Thanks for the explanation. I'd definitely open an issue to discuss this against Symfony.

However, your second message has me confused a bit. What you're saying is that since fixtures need to interact with entity objects (E.G. instantiate them for persistence), they are dependent on the bundle itself, right?

Well, in that sense, migrations are like that, too. Migrations are created to set up the environment for the persistence layer to function. For example, if you had two bundles, then having the migrations for both of those bundles in the `app/` directory rather than inside each bundle would be strange, right?

That's why I think both the migrations and fixtures should act the same. But let's see what the Symfony community at large has to say about this; maybe I'm having this all wrong.

Thanks for showing me the next step. You're great! :)

**Comment by tystr** [comment]
I typically use a bundleless structure for symfony2 apps, so having this bundle checking a global folder such as `app/` would be nice. Even better would be the ability to configure the paths in which this bundle should look for fixtures.

**Comment by weaverryan** [comment]
I think what needs to be done here is for some motivated person to do exactly what @tystr is asking for: add a configuration option for additional directories to scan when looking for fixtures.

**Comment by weaverryan** [comment]
Pff, I didn't even see #139, which would appear to fix this

**Comment by tystr** [comment]
@weaverryan I was just about to comment that I'd solved this in that PR :)
